src/Uno.Extensions.Validation/IValidatorTypedInstance.cs

```diff
@@ -0,0 +1,14 @@
+﻿namespace Uno.Extensions.Validation;
+
+internal interface IValidatorTypedInstance : IValidator
+{
+	Type InstanceType { get; }
+}
+
+internal record ValidatorTypedInstance<T>(IServiceProvider Services) : IValidatorTypedInstance
+{
+	public Type InstanceType => typeof(T);
+	public IValidator<T> Validator => Services.GetRequiredService<IValidator<T>>();
+	public ValueTask<IEnumerable<ValidationResult>> ValidateAsync(object instance, ValidationContext? context = null, CancellationToken cancellationToken = default) => Validator.ValidateAsync(instance, context, cancellationToken);
+}
```

Comment:
I would prefer this broken into 2 files
- IValidatorTypedInstance
- ValidatorTypedInstance{T}